index.bs

```diff
@@ -6014,6 +6061,85 @@ To <dfn data-lt="bidirectionally broadcasting the shapes">bidirectionally broadc
 |shapeA| is <dfn>bidirectionally broadcastable</dfn> to |shapeB| if [=bidirectionally broadcasting the shapes=] |shapeA| and |shapeB| does not result in failure.
 </p>
 
+## Casting ## {#algorithms-casting}
+
+Explicit numeric casting is used in algorithms where parameters passed as {{MLNumber}}, {{MLFiniteNumber}}, or {{double}} need to be converted to match the {{MLOperandDataType}} of input or output {{MLOperand}}s.
```

Comment:
I am wondering whether we can leverage or refer to ECMAScript numeric casting algorithms, say
- [`Math.fround()`](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-math.fround)
- [`Math.f16round()`](https://tc39.es/proposal-float16array/#sec-math.f16round)
- [`ToInt8()`](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-toint8)
- [`ToUint8()`](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-touint8)
- [`ToInt32()`](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-toint32)
- [`ToUint32()`](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-touint32)
- [`ToBigInt64()`](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-tobigint64)
- [`ToBigUint64()`](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-tobiguint64)

Reply:
@a-sully and I looked at this; the problem with those is that the input is a JS Number (i.e. float64) and they throw if anything else is passed (like a BigInt) due to their call to [ToNumber()](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-tonumber). Referencing those algorithms is also somewhat wrong given the [spec processing model](https://garykac.github.io/procspec/#processing-model) where within algorithms we're no longer dealing with JS types. 

Ideally we'd work with WebIDL and export the algorithms defined there, but we can do that later and generally we add stuff to WebIDL only when its used in multiple specs.

Reply:
> and they throw if anything else is passed (like a BigInt)

😢 I would love to see Javascript work more cleanly with 64-bit integers, as it's rather messy currently (compared to say C++ where casting between 64-bit integers and 64-bit floats works as expected) where casting in JS from 3.14159 to BigInt throws instead of just truncating o_o (meanwhile, the same language is happy to implicitly and silently cast between strings and numbers, and [sorting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) a list of numbers *actually* sorts your numbers *as if* they were strings 🙃).